dotnet/samples/KernelSyntaxExamples/Example75_AgentTools.cs

```diff
@@ -0,0 +1,168 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.SemanticKernel.Experimental.Agents;
+
+// ReSharper disable once InconsistentNaming
+/// <summary>
+/// Showcase usage of code_interpreter and retrieval tools.
+/// </summary>
+public static class Example75_AgentTools
+{
+    /// <summary>
+    /// Specific model is required that supports agents and parallel function calling.
+    /// Currently this is limited to Open AI hosted services.
+    /// </summary>
+    private const string OpenAIFunctionEnabledModel = "gpt-4-1106-preview";
+
+    // Track agents for clean-up
+    private static readonly List<IAgent> s_agents = new();
+
+    /// <summary>
+    /// Show how to combine coordinate multiple agents.
+    /// </summary>
+    public static async Task RunAsync()
+    {
+        Console.WriteLine("======== Example75_AgentTools ========");
+
+        if (TestConfiguration.OpenAI.ApiKey == null)
+        {
+            Console.WriteLine("OpenAI apiKey not found. Skipping example.");
+            return;
+        }
+
+        // Run agent with code-interpreter tool
+        await RunCodeInterpreterToolAsync();
+
+        // Run agent with retrieval tool
+        await RunRetrievalToolAsync();
+    }
+
+    private static async Task RunCodeInterpreterToolAsync()
+    {
+        Console.WriteLine("======== Run:CodeInterpreterTool ========");
+
+        var builder =
+            new AgentBuilder()
+                .WithOpenAIChatCompletion(OpenAIFunctionEnabledModel, TestConfiguration.OpenAI.ApiKey)
+                .WithInstructions("Write only code to solve the given problem without comment.");
+
+        try
+        {
+            var defaultAgent =
+                Track(
+                    await builder.BuildAsync());
+
+            var codeInterpreterAgent =
+                Track(
+                    await builder.WithCodeInterpreter().BuildAsync());
+
+            await ChatAsync(
+                defaultAgent,
+                codeInterpreterAgent,
+                "What is the solution to `3x + 2 = 14`?",
+                "What is the fibinacci sequence until 101?");
+        }
+        finally
+        {
+            await Task.WhenAll(s_agents.Select(a => a.DeleteAsync()));
+        }
+    }
+
+    private static async Task RunRetrievalToolAsync()
+    {
+        Console.WriteLine("======== Run:RunRetrievalTool ========");
+
+        // REQUIRED:
+        //
+        // Use `curl` to upload document prior to running example and assign the
```

Comment:
Once this PR is in, we could actually change this example to point to the travelinfo.txt from GitHub.

This way, curl won't be required anymore and this example can be turned into an integration test like the others...

Reply:
Good idea.